dplock/dplock.go

```diff
@@ -138,8 +148,24 @@ func (l *Lock) Acquire(ctx context.Context, id string) (lockID string, err error
 
 // Unlock releases an exclusive mongoDB lock for the provided id (if it exists)
 func (l *Lock) Unlock(ctx context.Context, lockID string) error {
-	_, err := l.Client.Unlock(ctx, lockID)
-	return err
+	retries := 0
+	for {
+		_, err := l.Client.Unlock(ctx, lockID)
+		if err == nil {
+			return nil
+		}
+		if retries >= UnlockMaxRetries {
+			return ErrUnlockMaxRetries
```

Comment:
Existing Apps that call this do not check the error return.
So, in the mean time i think we need to log and error between lines 157 & 158 so that the logs can be searched to see if there are issues with the unlock ... and then that gives reasons for further investigations as to what elsewhere may need looking into.